cpp/pagerank_module/algorithm_online/pagerank.cpp

```diff
@@ -122,14 +122,56 @@ void CreateRoute(const mg_graph::GraphView<> &graph, std::uint64_t start_id, std
 }
 
 ///
-///@brief Updates the context based on new edge addition. Reverts previous walks made from starting node and updates
+///@brief Creates a route starting from start_id, stores it in walk and updates the walk_index. Route is created via
+/// random walk depending on random number genrator.
+///
+///@param graph Graph for route creation
+///@param start_id Starting node in graph creation
+///@param walk Walk vector that stores a route
+///@param walk_index Index of a walk for context storing
+///@param epsilon Probability of stopping the route creation
+///
+void CreateRoute(const mgp::Graph &graph, std::uint64_t start_id, std::vector<std::uint64_t> &walk,
+                 std::uint64_t walk_index, double epsilon) {
+  std::uint64_t current_id = start_id;
+  while (true) {
+    std::vector<std::uint64_t> neighbor_ids;
+    for (const auto out_relationship : graph.GetNodeById(mgp::Id::FromUint(current_id)).OutRelationships()) {
+      neighbor_ids.push_back(out_relationship.To().Id().AsUint());
+    }
+    // auto neighbors = graph.Neighbours(current_id);
+
+    if (neighbor_ids.empty()) break;
+
+    // Pick and add a random outer relationship
+    auto next_id = neighbor_ids[GetRandInt(0, neighbor_ids.size())];
+    // auto next_id = neighbors[GetRandInt(0, number_of_neighbors)].node_id;
+    // next_id = graph.GetMemgraphNodeId(next_id);
+
+    walk.emplace_back(next_id);
+    pagerank_online_alg::context.walks_table[next_id].insert(walk_index);
+    pagerank_online_alg::context.walks_counter[next_id]++;
+
+    // Finish walk when random number is smaller than epsilon
+    // Average length of walk is 1/epsilon
+    if (GetRandFloat() < epsilon) {
+      break;
+    }
+
+    // current_id = graph.GetInnerNodeId(next_id);
+  }
+}
+
+///
+///@brief Updates the context based on new relationship addition. Reverts previous walks made from starting node and
+/// updates
 /// them.
 ///
 ///@param graph Graph for updating
-///@param new_edge New edge
+///@param new_relationship New relationship
 ///
-void UpdateCreate(const mg_graph::GraphView<> &graph, const std::pair<std::uint64_t, std::uint64_t> &new_edge) {
-  auto [from, to] = new_edge;
+void UpdateCreate(const mgp::Graph &graph, const std::pair<std::uint64_t, std::uint64_t> &new_relationship) {
+  auto [from, to] = new_relationship;
```

Comment:
Do you need a reference here too?
auto &[from, to] = new_relationship;

Reply:
Added